lib/array.js

```diff
@@ -108,7 +109,11 @@ internals.checkItems = function (items, wasArray, state, options) {
     var requireds = this._inner.requireds.slice();
     var inclusions = this._inner.inclusions.concat(requireds);
 
-    for (var v = 0, vl = items.length; v < vl; ++v) {
+    if (this._inner.ordered.length && !this._inner.items.length) {
```

Comment:
You'll have to find another way to do it, on each validation you'll be pushing that into the exclusions, that's kind of a leak.